core/src/validator.rs

```diff
@@ -168,7 +173,38 @@ impl Validator {
             config.storage_slots_per_turn,
             bank.slots_per_segment(),
         );
-
+        let bf = bank_forks.clone();
+        let counter = Arc::new(Mutex::new(0));
```

Comment:
Using an atomic would be nicer here actually:
```rust
let counter = Arc::new(AtomicUsize::new(0));
```
and then we can do this in the handler:
```rust
if counter.fetch_add(1, Ordering::Relaxed) > 0 {
  warn!("Validator aborting on second ^C");
  std::process::exit(1);
}
```.